src/uhs/sentinel/client.hpp

```diff
@@ -34,6 +34,12 @@ namespace cbdc::sentinel::rpc {
         /// \return true if initialization succeeded.
         auto init() -> bool;
 
+        /// Initializes the client. Establishes a connection to the sentinel.
+        /// \param error_fatal if false, the client will succeed initialization even if
+        /// the connection to the sentinel fails
+        /// \return true if initialization succeeded.
+        auto init(std::optional<bool> error_fatal) -> bool;
```

Comment:
Why is `error_fatal` in a `std::optional`? Is it just so you don't have to modify other parts of the code, or does it have a semantic meaning when it's `std::nullopt`?

Reply:
As a follow-up: if there is semantic meaning for `std::nullopt`, is there a reason to have this split into two overloads rather than defaulting the argument to `std::nullopt`?

Reply:
The semantic meaning to `std::nullopt` is to default back to the current behavior, which is to only treat errors as fatal when you're connecting to a cluster with one member. Otherwise it falls on the caller to implement this default, which means changes everywhere this is used. I have made this explicit in the doc comment.

I agree that I could just make this a default argument  to remove the explicit overload - so I have addressed that.